### ISSUE TITLE
Consumer - product alert emails are not translated

### ISSUE THREAD

**Issue by loic-paquin** [magento/magento2#35674]
### Preconditions and environment

- Magento version 2.4.4
- PHP 8.1
- Fix the issue  https://github.com/magento/magento2/issues/35602 since it will be executed first
- Fix also the issue  https://github.com/magento/magento2/issues/35673 since it will be executed first
- Install a language pack ( German , Fr ... doesn't matter)
-  Assign or create a store view with this language

### Steps to reproduce

- Enable product alert stock in BO (Configuration > Catalog >Product Alerts > Allow Alert When Product Comes Back in Stock > Yes)
- Create a customer on the new store view
- Select a Product without stock
- Register to a product stock alert
- Add stock to that product
- Launch the cron catalog_product_alert

### Expected result

- The consumer is launched
- The email is sent to the customer in the correct language

### Actual result

The email is sent in english even if the translations are available in the language pack

### Additional information

I was able to fix the problem by editing the class  _Magento\ProductAlert\Model\Email_ 

I added those lines after the call to the _startEnvironmentEmulation_ function  ( around L349)

 `  $area = $this->areaList->getArea(\Magento\Framework\App\Area::AREA_FRONTEND);`
`   $area->load(\Magento\Framework\App\Area::PART_TRANSLATE);`

( You will need to add _Magento\Framework\App\AreaList_ in the constructor)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Lima** [comment]
Hi @loic-paquin ,

Thanks for your contribution and collaboration.

As this issue related to this [issue](https://github.com/magento/magento2/issues/35602) https://github.com/magento/magento2/issues/35602 and it is taking time to resolve .
Kindly provide us any other way to reproduce the issue so that we can proceed further.

Thanks

**Comment by joseluisi4** [comment]
Reproduced issue in Magento pce 2.4.5

**Comment by dandrikop** [comment]
I have the same problem on Magento 2.4.6-p1.

**Comment by dandrikop** [comment]
@loic-paquin I didn't understand your comment that it will be needed to add Magento\Framework\App\AreaList in the constructor. Can you please provide the exact code changes?

**Comment by loic-paquin** [comment]
@dandrikop 

This is what I did at the time, I used a preference : 
https://gist.github.com/loic-paquin/d1511b52b9c7f34f87a58b06e26e5231

It worked well on 2.4.4, I cannot confirm it will works on current Magento version.

You can see I added on L69  AreaList $areaList in the constructor

**Comment by dandrikop** [comment]
@loic-paquin
It seems that the problem with translating the product alerts has been identified for Magento 2.4.7 (https://github.com/magento/magento2/tree/2.4.7-beta1/app/code/Magento/ProductAlert). At the updated **Product Alert** module, it seems that only the file app/code/Magento/ProductAlert/Model /Mailing/AlertProcessor.php has changed. I tried the updated ProductAlert module, but the unfortunately the problem with translations persisted. Of course, Magento 2.4.7 is still in beta phase, so maybe the problem is fixed till version 2.4.7 becomes stable.

Anyway, based on your directions, I created and applied the below patch and tested it on Magento 2.4.6-p1 with PHP 8.1.20, and it worked without any errors recorders in the log files. I would like to thank you for that.

`diff --git a/vendor/magento/module-product-alert/Model/Email.php b/vendor/magento/module-product-alert/Model/Email.php
index 379ae29..c6b1c12 100644
--- a/vendor/magento/module-product-alert/Model/Email.php
+++ b/vendor/magento/module-product-alert/Model/Email.php
@@ -31,6 +31,7 @@ use Magento\Store\Model\App\Emulation;
 use Magento\Store\Model\ScopeInterface;
 use Magento\Store\Model\StoreManagerInterface;
 use Magento\Store\Model\Website;
+use Magento\Framework\App\AreaList;
 
 /**
  * ProductAlert Email processor
@@ -139,6 +140,11 @@ class Email extends AbstractModel
      */
     protected $_customerHelper;
 
+    /**
+     * @var View
+     */
+    protected $areaList;
+
     /**
      * @param Context $context
      * @param Registry $registry
@@ -149,6 +155,7 @@ class Email extends AbstractModel
      * @param View $customerHelper
      * @param Emulation $appEmulation
      * @param TransportBuilder $transportBuilder
+     * @param AreaList $areaList;
      * @param AbstractResource $resource
      * @param AbstractDb $resourceCollection
      * @param array $data
@@ -164,6 +171,7 @@ class Email extends AbstractModel
         View $customerHelper,
         Emulation $appEmulation,
         TransportBuilder $transportBuilder,
+        AreaList $areaList,
         AbstractResource $resource = null,
         AbstractDb $resourceCollection = null,
         array $data = []
@@ -175,6 +183,7 @@ class Email extends AbstractModel
         $this->_appEmulation = $appEmulation;
         $this->_transportBuilder = $transportBuilder;
         $this->_customerHelper = $customerHelper;
+        $this->areaList = $areaList;
         parent::__construct($context, $registry, $resource, $resourceCollection, $data);
     }
 
@@ -346,7 +355,9 @@ class Email extends AbstractModel
         $storeId = (int) $this->getStoreId() ?: (int) $this->_customer->getStoreId();
         $store = $this->getStore($storeId);
 
-        $this->_appEmulation->startEnvironmentEmulation($storeId);
+        $this->_appEmulation->startEnvironmentEmulation($storeId, Area::AREA_FRONTEND, true);
+        $area = $this->areaList->getArea(\Magento\Framework\App\Area::AREA_FRONTEND);
+        $area->load(\Magento\Framework\App\Area::PART_TRANSLATE); 
 
         $block = $this->getBlock();
         $block->setStore($store)->reset();
`

**Comment by TheBadong** [comment]
Reproduced on 2.4.5-p2
[This solution](https://github.com/magento/magento2/issues/35674#issue-1283536512) works just fine. Does anyone know if it's been fixed in more recent versions ?

**Comment by nurullah** [comment]
> Since I am not getting product stock alert emails as this issue is dependent on the [issue](https://github.com/magento/magento2/issues/35602) . This will be tested after [issue](https://github.com/magento/magento2/issues/35602) got fixed. Hence, moving this issue to "On Hold".

Hello. It looks like dependent issue fixed. Could you please test this issue again? After my translations did not work, I realized that this was the problem.

**Comment by hoanxuanmai** [comment]
```php

use Magento\Framework\App\Area;
use Magento\Framework\App\ObjectManager;
use Magento\Framework\Phrase;
use Magento\Framework\Phrase\Renderer\Composite;
use Magento\Framework\Translate;
use Magento\ProductAlert\Model\Mailing\AlertProcessor;

class AlertProcessorPlugin
{
    function beforeProcess(AlertProcessor $object, string $alertType, array $customerIds, int $websiteId)
    {
        $objectManager = ObjectManager::getInstance();

        $objectManager->get(Translate::class)->loadData(Area::AREA_FRONTEND, true);

        Phrase::setRenderer($objectManager->get(Composite::class));
        return [$alertType, $customerIds, $websiteId];
    }
}

```
I just added this plugin to fix the product alert translation issue

it is working for me

**Comment by engcom-Dash** [comment]
Hi @loic-paquin 

We have reopened the dependent issue - [35602](https://github.com/magento/magento2/issues/35602). Will test this issue once the dependent issue is fixed.

Thanks.

**Comment by engcom-Dash** [comment]
Hi @loic-paquin 

As there is no activity on this issue for a long time, we believe the issue has been resolved, hence closing this issue.
Feel free to raise a new issue or reopen this if you need more assistance.

Thank you.

**Comment by ajung2** [comment]
Hello Magento-Team, 

we have also this translation problem in 2.4.6-p6 we use the working patch from @dandrikop from this site 
[https://community.magento.com/t5/Magento-2-x-Technical-Issues/Product-Alert-emails-are-not-translated/m-p/517435/highlight/true#M21150](url), man thank you for the share.

Hope it will be fixed in 2.4.7

Thank you.
